src/datasources/perf-ds/datasource.ts

```diff
@@ -316,6 +338,196 @@ export class OpenNMSDatasource {
+      if (!resourceIdPattern) { return result; }
+      const regexResourceId = new RegExp(resourceIdPattern);
+      if (!regexResourceId.test(resourceId)) { return result; }
+
+      _.forOwn(response.rrdGraphAttributes, (obj, id) => {
+        const attributePattern = properties.get(AllowedProperties.Attribute);
+        if (!attributePattern)  { return; }
+        const regex = new RegExp(attributePattern);
+        if (regex.test(id)) {
+          const match: Map<string, string> = new Map<string, string>();
+
+          match.set(AllowedProperties.ResourceId, OpenNMSDatasource.getRemoteResourceId(nodeId, resourceId));
+          match.set(AllowedProperties.Attribute, id);
+          result.push(match);
+        }
+      });
+    }
+    return result;
+  }
+
+  /**
+   * Depending if resource id or attribute is being checked
+   * @param sourceArray : source array from query
+   * @param types : allowed properties (resource id, attribute)
+   * @returns array of sources that containg glob expressions (* or |)
+   */
+  getGlobExpressionsOnly(sourceArray: any[]): Set<any> {
+    let globSources: Set<any> = new Set<any>();
+    let self = this;
+    _.each(sourceArray, (source) => {
+      _.forIn(AllowedProperties, (value, key) => {
+        let sourceType = self.getIdFromType(source, value);
+        if (OpenNMSGlob.hasGlob(sourceType) && !globSources.has(source)) {
+          globSources.add(source);
+          return;
+        }
+      });
+    });
+    return globSources;
+  }
+
+  /**
+   * Gets the Id value for the provided type
+   * @param source source object
+   * @param type property type (resourceId, attribute)
+   * @returns Id value
+   */
+  getIdFromType(source: any, type: string): string {
+    switch (type) {
+      case 'resourceId':
+        return OpenNMSDatasource.getResourceId(source[type]);
+        break;
+      case 'attribute':
+      default:
+        return source[type];
+        break;
+    }
+  }
+
+  /**
+   * Get a map with allowed properties to query and their corresponding regular expressions for search.
+   * @param source a single source object with search parameters
+   * @returns 
+   */
+  getGlobQueries(source: any): Map<AllowedProperties, string> {
+    const props: Map<AllowedProperties, string> = new Map<AllowedProperties, string>();
+    _.forIn(AllowedProperties, (value, key) => {
+      if (value === AllowedProperties.ResourceId){
+        props.set(value, OpenNMSGlob.getGlobAsRegexPattern(OpenNMSDatasource.getResourceId(source[value])));
+      }
+      else {
+        props.set(value, OpenNMSGlob.getGlobAsRegexPattern(source[value]));
+      }
+    });
+    return props;
+  }
+
+  /**
+   * Retrievd the node database id value from the resource id
+   * @param resource : resource id as <nodeId>.<resourceId>
+   * @returns Node Id 
+   */
+  static getNodeId(resource): string {
+    const matches = resource.match(/node(Source)?\[(.*)?\]\..*/);
```

Comment:
Like I mentioned in https://github.com/OpenNMS/opennms-helm/pull/395, you might want to make this not do a "greedy anything" match:
```suggestion
    const matches = resource.match(/node(Source)?\[([^\]*)?\]\..*/);
```